src/backend/distributed/commands/table.c

```diff
@@ -666,6 +697,95 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 					break;
 				}
 			}
+
+			/*
+			 * We check for ADD COLUMN .. DEFAULT expr
+			 * if expr contains nextval('user_defined_seq')
+			 * we should deparse the statement
+			 */
+			constraint = NULL;
+			foreach_ptr(constraint, columnConstraints)
+			{
+				if (constraint->contype == CONSTR_DEFAULT)
+				{
+					if (constraint->raw_expr != NULL)
+					{
+						ParseState *pstate = make_parsestate(NULL);
+						Node *expr = transformExpr(pstate, constraint->raw_expr,
+												   EXPR_KIND_COLUMN_DEFAULT);
+
+						if (contain_nextval_expression_walker(expr, NULL))
+						{
+							deparseAT = true;
+
+							ColumnDef *newColDef = copyObject(columnDefinition);
```

Comment:
Could use a comment on why we do this

Reply:
Actually, I don't know why I do this 😆 removing it